sklearn/linear_model/_omp.py

```diff
@@ -281,6 +282,18 @@ def _gram_omp(
         return gamma, indices[:n_active], n_active
 
 
+@validate_params(
+    {
+        "X": [np.ndarray],
```

Comment:
Since the first thing the function does is call check_array on X, X can be an array-like here.
```suggestion
        "X": ["array-like"],
```
Can you update the docstring accordingly ?

Reply:
Updated `linear_model.orthogonal_mp`